go/vt/sqlparser/sql.y

```diff
@@ -2556,6 +2561,70 @@ collate_opt:
     $$ = string($2)
   }
 
+default_keyword_opt:
+  {
+    $$ = false
+  }
+| DEFAULT
+  {
+    $$ = true
+  }
+
+charset_collate_opt:
+  {
+    $$ = nil
+  }
+| charset_collate
+  {
+    $$ = $1
+  }
+
+charset_collate:
+  charset_default_opt
+  {
+    $$ = []*CharsetAndCollate{$1}
+  }
+| collate_default_opt
+  {
+    $$ = []*CharsetAndCollate{$1}
+  }
+| charset_collate collate_default_opt
```

Comment:
This rule and the one below aren't quite right, this allows constructions like DEFAULT CHARACTER SET 'foo' CHARACTER SET BINARY CHARSET BINARY

Make sure you have tests for things that shouldn't parse as well

Reply:
Talked about this offline, turns out MySQL actually does allow such constructs, so LGTM